test/unified/runner.js

```diff
@@ -0,0 +1,366 @@
+'use strict';
+const expect = require('chai').expect;
+const path = require('path');
+const semver = require('semver');
+const MongoClient = require('../../src/mongo_client').MongoClient;
+
+class Version {
+  static clean(version) {
+    return semver.coerce(version).version;
+  }
+  static check(supportedSchemaVersions, schemaVersion) {
+    let satisfies = false;
+    supportedSchemaVersions.map(Version.clean).forEach(v => {
+      if (satisfies) return;
+      const result = semver.satisfies(v, `^${Version.clean(schemaVersion)}`);
+      if (result) satisfies = true;
+    });
+    return satisfies;
+  }
+  static validate(supportedSchemaVersions, spec) {
+    const check = Version.check(supportedSchemaVersions, spec.schemaVersion);
+    if (!check) throw new Error(`invalid schemaVersion: ${spec.description}`);
+  }
+}
+
+class Match {
+  static value(value, matcher, expectation) {
+    switch (matcher) {
+      // TODO: add in all matchers
+      case '$$unsetOrMatches': {
+        expect(value).to.be.oneOf([undefined, expectation]);
+      }
+    }
+  }
+  static expectedResult(expected, actual) {
+    if (typeof expected === 'object') {
+      Object.keys(expected).forEach(key => {
+        const expectedValue = expected[key];
+        const actualValue = actual[key];
+        if (typeof expectedValue === 'object') {
+          Object.keys(expectedValue).forEach(matcher => {
+            const matcherValue = expectedValue[matcher];
+            Match.value(actualValue, matcher, matcherValue);
+          });
+        }
+      });
+    }
+  }
+}
+
+// TODO: create constants for all client event ids
+const COMMAND_STARTED = 'commandStarted';
+
+// TODO: add in mappings for all methods
+const argMap = {
+  insertOne: a => [a.document]
+};
+
+class Runner {
+  constructor(spec, filePath) {
+    this.supportedSchemaVersions = ['1.0.0'];
+    Version.validate(this.supportedSchemaVersions, spec);
+    if (filePath) {
+      this.filePath = filePath;
+      this.fileName = path.basename(filePath);
+    }
```

Comment:
What are these used for? I don't see them referred to elsewhere in the PR

Reply:
This isn't used, can remove, but also can be useful for debugging when adding in multiple files.